src/main/java/com/unstoppabledomains/resolution/ENS.java

```diff
@@ -0,0 +1,80 @@
+package com.unstoppabledomains.resolution;
+
+import java.util.Arrays;
+
+import com.unstoppabledomains.exceptions.NamingServiceException;
+import com.unstoppabledomains.resolution.artifacts.Numeric;
+import com.unstoppabledomains.resolution.contracts.Contract;
+import com.unstoppabledomains.resolution.contracts.ens.EnsContractType;
+import com.unstoppabledomains.resolution.contracts.ens.Registry;
+import com.unstoppabledomains.resolution.contracts.ens.Resolver;
+
+public class ENS extends NamingService {
+
+  static final String REGISTRY_ADDRESS = "0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e";
+  private final String providerURL;
+  private final Registry registryContract;
+  
+
+  public ENS(String blockchainProviderURL) {
+    super();
+    this.providerURL = blockchainProviderURL;
+    this.registryContract = (Registry)buildContract(REGISTRY_ADDRESS, EnsContractType.Registry);
+  }
+
+  @Override
+  Boolean isSupported(String domain) {
```

Comment:
In my opinion, we should make those methods either `public` or `protected` here and in `NamingService`. I thin `protected` would be more preferable since we have `Resolution` as an entry point for our functionally.

Reply:
It also makes sense to convert `NamingService` to an interface and use `NameHash#nameHash(String domain)` directly for a better concept distinguishment.